tests/qat_after_ptq/torch/test_qat_classification.py

```diff
@@ -0,0 +1,236 @@
+    )
+    calibration_dataset = nncf.Dataset(val_data_loader_batch_one, transform_fn)
+    return DatasetSet(
+        train_data_loader=train_data_lodaer,
+        val_data_loader=val_data_loader,
+        train_sampler=train_sampler,
+        calibration_dataset=calibration_dataset,
+    )
+
+
+def test_compression_training(quantization_config_path: Path, data_dir: str, mocker):
+    if "imagenet" not in quantization_config_path.stem:
+        pytest.skip("Test works only with imagenet models by far.")
+
+    nncf_config = NNCFConfig.from_json(quantization_config_path)
+    if "compression" not in nncf_config or nncf_config["compression"]["algorithm"] != "quantization":
+        pytest.skip("Config without compression")
+
+    config = get_sample_config(quantization_config_path, data_dir)
+    if "accuracy_aware_training" in config:
+        pytest.skip("Accuracy Aware training is not supported yet for QAT with PTQ.")
+
+    if "pretrained" not in config or not config["pretrained"]:
+        pytest.skip("Test supports only pretrained models.")
+
+    start_worker(main_worker, config)
+
+
+def main_worker(current_gpu: int, config: SampleConfig):
+    configure_device(current_gpu, config)
+    if is_main_process():
+        configure_logging(logger, config)
+    else:
+        config.tb = None
+
+    pretrained = is_pretrained_model_requested(config)
+    model_name = config["model"]
+    # create model
+    logger.info(f"\nCreating model from config: {config.config}")
+    model = load_model(
+        model_name,
+        pretrained=pretrained,
+        num_classes=config.get("num_classes", 1000),
+        model_params=config.get("model_params"),
+        weights_path=config.get("weights"),
+    )
+    model.to(config.device)
+
+    datasets = get_datasets(config)
+    criterion = nn.CrossEntropyLoss()
+    criterion = criterion.to(config.device)
+
+    logger.info("Original model validation:")
+    original_accuracy, *_ = validate(datasets.val_data_loader, model, criterion, config)
+
+    logger.info("Apply quantization to the model:")
+    config_quantization_params = config["compression"]
+
+    preset = get_quantization_preset(config_quantization_params)
+    advanced_parameters = get_advanced_ptq_parameters(config_quantization_params)
+    subset_size = get_num_samples(config_quantization_params)
+
+    quantized_model = nncf.quantize(
+        model,
+        datasets.calibration_dataset,
+        preset=preset,
+        advanced_parameters=advanced_parameters,
+        subset_size=subset_size,
+    )
+
+    acc_drop = train(
+        quantized_model,
+        config,
+        criterion,
+        datasets,
+        original_accuracy,
+        get_mocked_compression_ctrl(),
+    )
+    assert accuracy_drop_is_acceptable(acc_drop)
+
+
+def accuracy_drop_is_acceptable(acc_drop: float) -> bool:
+    """
+    Returns True in case acc_drop is less than 1 precent.
+    """
+    return acc_drop < 1.0
+
+
+def catch_nans_wrapper(fn):
```

Comment:
why not to use it as a test that the training is correct ?

Reply:
I made a separate check, please see